### ISSUE TITLE
Questions on how to train the model in another framework

### ISSUE THREAD

**Issue by SebastienTs** [deepang-ai/Slim-UNETR#3]
I'm trying to train your model in a different framework with a standard optimizer (Adam) and loss (cross-entropy). 
The training completes but so far the results are quite worse than a default UNET I use as reference. 

I don't know if it is because the default hyper-paremeters are not optimal but the dataset is rather simple (it's more of a test dataset). Could you give me some guidelines on what optimizer and loss to use and how you would adjust the hyperparameters? 

Also, do you know of another implementation which would allow to change the number of encoder and decoder stages?

**Comment by SebastienTs** [comment]
I managed to improve the results, the issue was that the downsampling at the first stage (4) was too aggressive, setting it to 2 helped a lot to improve the results. I believe that it is possible to disable a stage by setting its number of blocks, heads and r to 0. Also, the stage 4 seems unique and its width is controlled by embed_dim (power of 2) and embedding_dim must be set to 64. I will read the article more in details to further understand the implementation!
